app/controllers/search_controller.rb

```diff
@@ -0,0 +1,17 @@
+class SearchController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_query_params
+
+  def users
+    distance = @query_params[:distance] == 'All' ? '' : @query_params[:distance]
+    @users = User.search_by_interest(@query_params[:interest])
+                 .search_by_skill(@query_params[:skill])
+                 .search_by_distance(current_user, distance)
+  end
+
+  private
+
+  def set_query_params
```

Comment:
Don't believe you need these permits because the params are not used in a call to an active record model method like `new` or `create` or `update`.